packages/block-editor/src/components/block-visibility/toolbar.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { ToolbarButton, ToolbarGroup } from '@wordpress/components';
+import { useRef, useEffect } from '@wordpress/element';
+import { seen, unseen } from '@wordpress/icons';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { cleanEmptyObject } from '../../hooks/utils';
+
+export default function BlockVisibilityToolbar( { clientIds } ) {
+	const blocks = useSelect(
+		( select ) => {
+			return select( blockEditorStore ).getBlocksByClientId( clientIds );
+		},
+		[ clientIds ]
+	);
+	const canToggleBlockVisibility = useSelect(
+		( select ) => {
+			const { getBlockName, getBlocksByClientId } =
+				select( blockEditorStore );
+			const _blocks = getBlocksByClientId( clientIds );
+			return _blocks.every( ( { clientId } ) =>
+				hasBlockSupport(
+					getBlockName( clientId ),
+					'blockVisibility',
+					true
+				)
+			);
+		},
+		[ clientIds ]
+	);
```

Comment:
The `useSelect` calls are both pretty similar, they could be combined into one to reduce subscriptions, or you could use `blocks` from the first `useSelect` in the second.

Reply:
Good catch, fixed in dd0d1cb4dfaf80a38c9db55c9a3f6f1fb9774460